### PR TITLE
Feature1 - Storing Movie Details

### PR DESCRIPTION
**Pull request by ghost**
Added Storing Movie Details Feature

### CODE REVIEW

**Comment by RiyasAhamed21** [comment]
- Added Storing Movie Details Deature

File: src/main/java/in/riyasahamed/service/MovieService.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import in.riyasahamed.model.Movie;
+import in.riyasahamed.validator.MovieValidator;
+
+public class MovieService {
+	
+	private MovieService() {
+		// Default Constructor
+	}
+
+	private static List<Movie> movies=new ArrayList<>();
```
Comment (csys-fresher-batch-2021) [inline src/main/java/in/riyasahamed/service/MovieService.java]: Add final

File: src/main/java/in/riyasahamed/service/MovieService.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	 * @param actor
+	 * @param rating
+	 */
+	public static void addMovieDetails(String name, String actor, Float rating) {
+		try {
+			MovieValidator.validateMovieDetails(name, actor, rating);
+			Movie movie = new Movie(name, actor, rating);
+			movies.add(movie);
+		} catch (Exception e) {
+			throw new ValidationException(e.getMessage());
```
Comment (csys-fresher-batch-2021) [inline src/main/java/in/riyasahamed/service/MovieService.java:32]: Add e.printStackTrace();

File: src/main/java/in/riyasahamed/servlets/AddMovieServlet.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+	/**
+	 * @see HttpServlet#doGet(HttpServletRequest request, HttpServletResponse response)
+	 */
+    @Override
+	protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+		String movie=request.getParameter("movie");
+		String actor=request.getParameter("actor");
+		Float rating=Float.parseFloat(request.getParameter("rating"));
```
Comment (csys-fresher-batch-2021) [inline src/main/java/in/riyasahamed/servlets/AddMovieServlet.java]: Move the parsing code inside try block

File: src/main/java/in/riyasahamed/validator/MovieValidator.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+	}
+	
+	/**
+	 * This Method Validates the Movie Details
+	 * @param name
+	 * @param actor
+	 * @param rating
+	 */
+	public static void validateMovieDetails(String name, String actor, Float rating) {
+		if (name == null || name.trim().equals("")|| name.equalsIgnoreCase("null")) {
```
Comment (csys-fresher-batch-2021) [inline src/main/java/in/riyasahamed/validator/MovieValidator.java]: Remove - name.equalsIgnoreCase("null")

File: src/main/java/in/riyasahamed/validator/MovieValidator.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+			throw new InvalidMovieNameException("Invalid Movie Name");
+		}
+		else if(actor == null || actor.trim().equals("")) {
+			throw new InvalidActorNameException("Invalid actor Name");
+		}
+		else if(rating < 0 || rating > 10 ) {
+			throw new InvalidRatingException("Invalid Rating");
```
Comment (csys-fresher-batch-2021) [inline src/main/java/in/riyasahamed/validator/MovieValidator.java]: Suggestion : "Invalid Rating. Valid Values are 1 to 10."

File: src/main/webapp/addMovie.jsp [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+<title>Add Movie</title>
+</head>
+<body>
+	<jsp:include page="header.jsp"></jsp:include>
+	<main class="container-fluid">
+		<h3>Add Movie</h3>
+		<form action="AddMovieServlet">
+
+				<br> <label for="name">Movie Name :</label> 
+				<input type="text" name="movie" id="name" autocomplete="on" required placeholder="Enter Movie Name" autofocus><br /> <br>
```
Comment (csys-fresher-batch-2021) [inline src/main/webapp/addMovie.jsp]: Use autocomplete="off"

File: src/main/webapp/addMovie.jsp [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+			<button class="btn btn-secondary" type="submit">Add Movie</button>
+			<button class="btn btn-danger" type="reset">Reset</button>
+			<br />
+		</form>
+		
+		<%
+		String errorMessage = request.getParameter("errorMessage");
+		if (errorMessage != null) {
+			out.println("<font color='red'>" + errorMessage + "</font>");
+		}
```
Comment (csys-fresher-batch-2021) [inline src/main/webapp/addMovie.jsp]: Move Message block above the form.

File: src/main/webapp/addMovie.jsp [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+				<label for="name">Actor:</label> 
+				<input type="text" name="actor" id="actor" autocomplete="on" required placeholder="Enter Actor Name"><br />
+				<br><label for="name">Rating:</label> 
+				<input type="number" name="rating" id="rating" autocomplete="on" required placeholder="Enter Rating"><br /> <br>
+			<button class="btn btn-secondary" type="submit">Add Movie</button>
+			<button class="btn btn-danger" type="reset">Reset</button>
+			<br />
+		</form>
+		
+		<%
```
Comment (csys-fresher-batch-2021) [inline src/main/webapp/addMovie.jsp]: Suggestion: Move this code to reusable file. "message.jsp"

File: src/main/webapp/index.jsp [hunk 9]
```diff
@@ -1,20 +1,15 @@
 <%@page import="java.util.Date"%>
 <%@page import="java.time.LocalDate"%>
```
Comment (csys-fresher-batch-2021) [inline src/main/webapp/index.jsp]: Remove unused imports

File: src/test/java/in/riyasahamed/validator/MovieValidatorTest.java [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+
+public class MovieValidatorTest {
+
+	/**
+	 * This test case is for valid Movie Details
+	 */
+	@Test
+	public void testValidMovieDetails() {
+		try {
+			MovieService.addMovieDetails("Master", "Vijay", (float) 8.5);
```
Comment (csys-fresher-batch-2021) [inline src/test/java/in/riyasahamed/validator/MovieValidatorTest.java:19]: assertTrue(true);

File: src/test/java/in/riyasahamed/validator/MovieValidatorTest.java [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+
+	}
+
+	/**
+	 * This test case is for Invalid Movie Details
+	 */
+	@Test
+	public void testinvalidMovieDetails() {
+		try {
+			MovieService.addMovieDetails(null, "vijay", (float) 8.0);
```
Comment (csys-fresher-batch-2021) [inline src/test/java/in/riyasahamed/validator/MovieValidatorTest.java:34]: fail();

File: src/test/java/in/riyasahamed/validator/MovieValidatorTest.java [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+		} catch (RuntimeException e) {
+			e.printStackTrace();
```
Comment (csys-fresher-batch-2021) [inline src/test/java/in/riyasahamed/validator/MovieValidatorTest.java]: Remove e.printStackTrace()

**Review by RiyasAhamed21 — APPROVED** [review]
LGTM

**Pull request merged**